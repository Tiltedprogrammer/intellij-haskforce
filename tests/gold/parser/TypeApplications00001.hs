module TypeApplications00001 where


foo :: ((Maybe Int))

foo = bar x X baz $ \spam -> eggs

foo = bar @_ @X baz $ \spam -> eggs
