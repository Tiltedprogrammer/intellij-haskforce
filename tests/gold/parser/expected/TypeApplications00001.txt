Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('TypeApplications00001')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    GENDECL
      VARS
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('foo')
      PsiElement(HaskellTokenType.::)('::')
      CTYPE
        TYPEE
          ATYPE
            PsiElement(HaskellTokenType.()('(')
            CTYPE
              TYPEE
                ATYPE
                  PsiElement(HaskellTokenType.()('(')
                  CTYPE
                    TYPEE
                      ATYPE
                        OQTYCON
                          QTYCON
                            TYCON
                              CONID
                                PsiElement(HaskellTokenType.conidRegexp)('Maybe')
                      ATYPE
                        OQTYCON
                          QTYCON
                            TYCON
                              CONID
                                PsiElement(HaskellTokenType.conidRegexp)('Int')
                  PsiElement(HaskellTokenType.))(')')
            PsiElement(HaskellTokenType.))(')')
    FUNORPATDECL
      PAT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('foo')
      RHS
        PsiElement(HaskellTokenType.=)('=')
        EXP
          QVAR
            QVARID
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('bar')
          QVAR
            QVARID
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('x')
          QCON
            QCONID
              CONID
                PsiElement(HaskellTokenType.conidRegexp)('X')
          QVAR
            QVARID
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('baz')
          QOP
            QVAROP
              QVARSYM
                VARSYM
                  PsiElement(HaskellTokenType.$)('$')
          PsiElement(HaskellTokenType.\)('\')
          VARID
            PsiElement(HaskellTokenType.varidRegexp)('spam')
          PsiElement(HaskellTokenType.->)('->')
          EXP
            QVAR
              QVARID
                VARID
                  PsiElement(HaskellTokenType.varidRegexp)('eggs')
    FUNORPATDECL
      PAT
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('foo')
      RHS
        PsiElement(HaskellTokenType.=)('=')
        EXP
          QVAR
            QVARID
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('bar')
          PsiElement(HaskellTokenType.@)('@')
          PsiElement(HaskellTokenType._)('_')
          PsiElement(HaskellTokenType.@)('@')
          CTYPE
            TYPEE
              ATYPE
                OQTYCON
                  QTYCON
                    TYCON
                      CONID
                        PsiElement(HaskellTokenType.conidRegexp)('X')
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('baz')
          QOP
            QVAROP
              QVARSYM
                VARSYM
                  PsiElement(HaskellTokenType.$)('$')
          PsiElement(HaskellTokenType.\)('\')
          VARID
            PsiElement(HaskellTokenType.varidRegexp)('spam')
          PsiElement(HaskellTokenType.->)('->')
          EXP
            QVAR
              QVARID
                VARID
                  PsiElement(HaskellTokenType.varidRegexp)('eggs')